zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -127,26 +127,39 @@ impl<F: Field> ExecutionGadget<F> for ErrorOOGCallGadget<F> {
         let opcode = step.opcode().unwrap();
         let is_call_or_callcode =
             usize::from([OpcodeId::CALL, OpcodeId::CALLCODE].contains(&opcode));
-        let [tx_id, is_static] =
-            [0, 1].map(|index| block.get_rws(step, index).call_context_value());
-        let [gas, callee_address] = [2, 3].map(|index| block.get_rws(step, index).stack_value());
+
+        let mut rws = StepRws::new(block, step);
+
+        let tx_id = rws.next().call_context_value();
+        let is_static = rws.next().call_context_value();
+
+        let gas = rws.next().stack_value();
+        let callee_address = rws.next().stack_value();
+
         let value = if is_call_or_callcode == 1 {
-            block.get_rws(step, 4).stack_value()
+            rws.next().stack_value()
         } else {
             U256::zero()
         };
-        let [cd_offset, cd_length, rd_offset, rd_length] =
-            [4, 5, 6, 7].map(|i| block.get_rws(step, is_call_or_callcode + i).stack_value());
 
-        let callee_code_hash = block
-            .get_rws(step, 9 + is_call_or_callcode)
-            .account_codehash_pair()
-            .0;
+        if is_call_or_callcode == 1 {
+            rws.offset_sub(1);
+        }
+
+        rws.offset_add(is_call_or_callcode);
```

Comment:
Isn't the following block unnecessary?
```
if is_call_or_callcode == 1 {
    rws.offset_sub(1);
}

rws.offset_add(is_call_or_callcode);
```

If `is_call_or_callcode == 1` you first decrease by 1 and then increase by 1.
If `is_call_or_callcode == 0` you don't decrease by 1 (stays the same) and then increase by 0 (stays the same).

Reply:
My bad 😅  looks like some debugging residue, Good catch!